File: src/main/java/io/github/hadixlin/iss/win/WinNative.java
package io.github.hadixlin.iss.win;

import com.sun.jna.Native;
import com.sun.jna.platform.win32.User32;
import com.sun.jna.win32.W32APIOptions;

public interface WinNative extends User32 {
    WinNative INSTANCE = (WinNative) Native.loadLibrary("user32", WinNative.class, W32APIOptions.DEFAULT_OPTIONS);

     HANDLE GetKeyboardLayout(DWORD pid);


}

File: src/main/kotlin/io/github/hadixlin/iss/win/WinNative.kt
package io.github.hadixlin.iss.win

import com.sun.jna.Native
import com.sun.jna.platform.win32.User32
import com.sun.jna.platform.win32.WinDef
import com.sun.jna.platform.win32.WinNT
import com.sun.jna.win32.W32APIOptions

/**
 * @author hadix
 */
interface WinNative : User32 {

    fun GetKeyboardLayout(pid: WinDef.DWORD): WinNT.HANDLE

    companion object {
        val INSTANCE = Native.load(
            "user32",
            WinNative::class.java,
            W32APIOptions.DEFAULT_OPTIONS
        ) as WinNative
    }

}
